framework/base/Model.php

```diff
@@ -441,7 +447,7 @@ public function createValidators()
         foreach ($this->rules() as $rule) {
             if ($rule instanceof Validator) {
                 $validators->append($rule);
-            } elseif (is_array($rule) && isset($rule[0], $rule[1])) { // attributes, validator type
+            } elseif (is_array($rule) && array_key_exists(0, $rule) && isset($rule[1])) { // attributes, validator type
```

Comment:
To support adding rules like that:

```php
[null, CommonValidator::className()]
```

or

```php
[null, `validateCommon`]
```

We can't leave `isset($rule[0])` check as is because it will always give `false` value for `null`s and as a result exception will be thrown that validation rule needs both attribute(s) and validatior to be specified.

From the other side, omitting first parameter completely is very ambiguous.

With `array_key_exists` we are making sure that it just exists, but it can be `null`.

`null` means that this rule does not directly related with specific attribute(s).